### PR TITLE
Replicated fetches timeouts

### PR DESCRIPTION
**Pull request by nvartolomei**
The test is a bit lousy. Wrote it to confirm that timeouts work and achieve the behavior I was looking for.

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Allow customizing timeouts for http connections used for replication independently from other http timeouts.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by nvartolomei** [comment]
Why would it timeout at `DROP TABLE`? 🤔

**Comment by nvartolomei** [comment]
```
2021.02.05 01:06:38.857000 [ 36 ] {} <Error> DatabaseCatalog: Cannot drop table default.t (bb18f622-0fc3-464f-b653-0b358c2f1c20). Will retry later.: Code: 242, e.displayText() = DB::Exception: Can't drop readonly replicated table (need to drop data in ZooKeeper as well)
```

It seems then ZK connections did timeout and never reconnected after partition manager "healed" the network.

File: tests/integration/test_replicated_fetches_network_partition/test.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    with PartitionManager() as pm:
+        # Make node1 very slow, node3 should replicate from node2 instead.
+        pm.add_network_delay(node1, 2000)
+
+        # node3 starts to replicate from node 1
+        node3.query("SYSTEM START FETCHES t")
+
+        # Wait some time to give a chance for node3 to try replicating without success from node1.
+        time.sleep(10)
```
Comment (KochetovNicolai) [inline tests/integration/test_replicated_fetches_network_partition/test.py:65]: Long sleeps in tests makes me sad :(

Can we create a small loop, like below, which reads `system.replication_queue` and checks if it has fetch with exception?

File: tests/integration/test_replicated_fetches_network_partition/test.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+            parts_fetched = node3.query("SELECT count() FROM system.parts WHERE table = 't'").strip()
+            print('parts_fetched:', parts_fetched)
+            print()
+
+            # Replication done.
+            if parts_fetched == "5":
+                break
+
+            time.sleep(3)
```
Comment (KochetovNicolai) [inline tests/integration/test_replicated_fetches_network_partition/test.py:83]: Sleep in a loop is much better. But I would rather use small timeout, like `0.1`.
Also, maybe we can use infinite loop and throw if, like, 10 seconds passed

File: tests/integration/test_replicated_fetches_network_partition/test.py (hunk 1 above)
Comment (nvartolomei) [reply tests/integration/test_replicated_fetches_network_partition/test.py:65]: That would be much better indeed.

File: tests/integration/test_replicated_fetches_network_partition/test.py (hunk 2 above)
Comment (nvartolomei) [reply tests/integration/test_replicated_fetches_network_partition/test.py:83]: Infinite loop sounds better, my bad that I copy/pasted this from another test (...replication limits...).
Not sure if 10 seconds are enough for all tests environments, CI sometimes is slow. I think it should be fine to loop for the test timeout 300s in the worst case...

Comment (KochetovNicolai) [reply tests/integration/test_replicated_fetches_network_partition/test.py:83]: 300s is also ok

File: src/Storages/MergeTree/MergeTreeSettings.h [hunk 3]
```diff
@@ -80,6 +80,9 @@ struct Settings;
     M(UInt64, replicated_max_parallel_fetches_for_host, DEFAULT_COUNT_OF_HTTP_CONNECTIONS_PER_ENDPOINT, "Limit parallel fetches from endpoint (actually pool size).", 0) \
     M(UInt64, replicated_max_parallel_sends, 0, "Limit parallel sends.", 0) \
     M(UInt64, replicated_max_parallel_sends_for_table, 0, "Limit parallel sends for one table.", 0) \
+    M(Seconds, replicated_fetches_http_connection_timeout, 0, "HTTP connection timeout for part fetch requests. Inherited from default profile `http_connection_timeout` if not set explicitly.", 0) \
```
Comment (nvartolomei) [inline src/Storages/MergeTree/MergeTreeSettings.h:83]: Can someone please double-check whether this is true? I tried to re-read the code, and it seems that there is no way to actually change timeouts in a non-query context.

**Review by nvartolomei — COMMENTED** [review]
Rewrote the test. Let's see how CI likes it.

File: src/Storages/MergeTree/MergeTreeSettings.h (hunk 3 above)
Comment (alexey-milovidov) [reply src/Storages/MergeTree/MergeTreeSettings.h:83]: It should be possible to change it in default profile (or in system profile if it is set).
I did not check it, though.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**